### PR TITLE
KFLUXBUGS-24: bump dev/stage overlay chains deployment to use performance arguments

### PR DESCRIPTION
**Pull request by gabemontero**
I'm not sure if these new args are just ignored, or rejected as invalid, if we don't have a version of chains installed that actually processes them.  We should find out with the e2e's, as I've updated the developer overlay with this change

But if it does cause a failure, then getting the correct openshift pipelines version will serve as the verification that everything is correct

FYI I duplicated the policy we use for core tekton controller, which comes from https://github.com/tektoncd/pipeline/blob/main/docs/tekton-controller-performance-configuration.md 

@lcarva @enarha @rupalibehera  FYI / PTAL

rh-pre-commit.version: 2.2.0
rh-pre-commit.check-secrets: ENABLED

### CODE REVIEW

**Review by lcarva — APPROVED** [review]

**Comment by enarha** [comment]
/lgtm
I suggest we wait with that PR until the nightly bump is merged (still in CI). Otherwise we risk getting conflicts with the deploy.yaml changes.

**Comment by gabemontero** [comment]
> /lgtm I suggest we wait with that PR until the nightly bump is merged (still in CI). Otherwise we risk getting conflicts with the deploy.yaml changes.

hold placed on this PR (I think you meant to say "this PR" when you said "that PR" @enarha )

**Comment by gabemontero** [comment]
@lcarva @enarha @rupalibehera - turns out the runtime errors out on unrecognized arguments.

Yaml snippet from the deployment with the args on the container look as specified:

```
                   "spec": {
                        "containers": [
                            {
                                "args": [
                                    "--threads-per-controller=32",
                                    "--kube-api-qps=50",
                                    "--kube-api-burst=50"
                                ],

```

but the konflux bootstrap fails with pipeline service not being ready, and on the chains pod, the container errors out, the logs have

```
flag provided but not defined: -threads-per-controller
Usage of /usr/local/bin/openshift-pipelines-chains-controller:
  -namespace string
    	Namespace to restrict informer to. Optional, defaults to all namespaces.

```

So this error no longer occurring will be the indicator that development is at the correct operator level.

/hold cancel

e2e will pass once @enarha PRs have merged.

**Comment by gabemontero** [comment]
I've rebased PR @enarha @lcarva 

let's see if chains comes up with the new arguments

**Comment by gabemontero** [comment]
chains came up fine now @lcarva @enarha with the new args

hit a seemingly unrelated multi platform build e2e flake

/retest

**Review by enarha — COMMENTED** [review]
/lgtm

**Comment by gabemontero** [comment]
again with the multi platform build test

```
 Summarizing 1 Failure:
  [TIMEDOUT] [multi-platform-build-service-suite Multi Platform Controller E2E tests] ibm power pc dynamic allocation when the Component with multi-platform-build is created [It] The multi platform secret is populated [multi-platform, ibmp-dynamic]
  /tmp/tmp.zoJi1i6gzA/tests/build/multi-platform.go:408
Ran 487 of 629 Specs in 5353.496 seconds
FAIL! - Suite Timeout Elapsed -- 486 Passed | 1 Failed | 11 Pending | 131 Skipped
```

I'm wondering if we can force merge @enarha @lcarva  ... I'll ask build team about it in their channel.

fyi  @hugares not ready just yet but might need a force merge

**Comment by lcarva** [comment]
As per slack discussion, the test failures appear to be unrelated. Given the urgency of these changes, let's just merge this.

**Pull request merged**
